File: app/src/main/java/com/moldedbits/genesis/models/response/PassageDetails.java
package com.moldedbits.genesis.models.response;

import java.util.List;

import lombok.Data;

@Data
public class PassageDetails {

    private String difficulty;

    private DisplayName displayName;

    private PassageText passageText;

    private List<Questions> questions;

    private List<Sentence> sentences;

    @Data
    public static class DisplayName {
        private String english;
        private String spanish;
    }

    @Data
    public static class PassageText {
        private String english;
        private String spanish;
    }

    @Data
    public static class Questions {
        private String answerText;

        private QuestionText questionText;
        private String type;
        private List<String> options;
    }

    @Data
    public static class QuestionText {
        private String english;
        private String spanish;
    }

    @Data
    public static class Sentence {
        private String english;
        private String spanish;
    }
}

File: app/src/main/java/com/moldedbits/genesis/models/response/PassageDetails.kt
package com.moldedbits.genesis.models.response


data class PassageDetails(val difficulty: String = "",
                          val displayName: TranslatableString = TranslatableString(),
                          val passageText: TranslatableString = TranslatableString(),
                          val questions: List<Question> = emptyList(),
                          val sentences: List<TranslatableString> = emptyList()
)

data class TranslatableString(val english: String = "", val spanish: String = "")

data class Question(val answerText: String = "",
                    val questionText: TranslatableString = TranslatableString(),
                    val type: String = "",
                    val options: List<String> = emptyList())